tests/backends/test_control.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Tests for the :mod:`aiida.backends.control` module."""
+import pytest
+
+from aiida.manage.manager import get_manager
+
+
+class MockRepositoryBackend():
+    """Mock of the RepositoryBackend for testing purposes."""
+
+    # pylint: disable=no-self-use
+
+    def get_info(self, *args, **kwargs):
+        """Method to return information."""
+        return 'this is information about the repo'
+
+    def delete_objects(self, *args, **kwargs):
+        """Method to delete objects."""
+
+    def maintain(self, full=False, **kwargs):
+        """Method to perform maintainance operations."""
+        text = 'this is a report on the maintenance procedure'
+        if full:
+            text += ' with the full flag on'
+        return text
+
+
+@pytest.fixture(scope='function')
+@pytest.mark.usefixtures('clear_database_before_test')
+def clear_storage_before_test():
+    """Clears the storage before a test."""
+    #-----------------------------------------------------------------#
+    # For some reason `clear_database_before_test` is not cleaning out
+    # the nodes so I have to do it manually.
+
+    from aiida import orm
+    from aiida.tools import delete_nodes
+    query = orm.QueryBuilder().append(orm.Node, project=['id'])
+    delete_nodes(query.all(flat=True), dry_run=False)
+
+    # Also would be good to disable the logging in delete_nodes
+    #-----------------------------------------------------------------#
+
+    repository = get_manager().get_backend().get_repository()
+    object_keys = list(repository.list_objects())
+    repository.delete_objects(object_keys)
+    repository.maintain(full=True)
+
+
+@pytest.mark.usefixtures('clear_database_before_test', 'clear_storage_before_test')
+def test_get_unreferenced_keyset():
```

Comment:
you are already using `clear_database_before_test` via `clear_storage_before_test`

```suggestion
@pytest.mark.usefixtures('clear_storage_before_test')
def test_get_unreferenced_keyset():
```